app/server/src/routes/routes.ts

```diff
@@ -203,20 +205,46 @@ export const apiEndpoints = (config => ({
         });
     },
 
-    async getProject(request, response) {
-        const {projectHash} = request.params;
-        await axios.get(`${config.api_url}/project/${projectHash}`)
-            .then(res => response.send(res.data))
-            .catch(function (error) {
-                sendError(response, error);
-            });
+    async getProject(request, response, next) {
+        await asyncHandler(next, async () => {
+            const {projectId} = request.params;
+            const {redis} = request.app.locals;
+            const store = userStore(redis);
+            const projectHash = await store.getProjectHash(request, projectId);
+            const res = await axios.get<APIResponse<ProjectResponse>>(`${config.api_url}/project/${projectHash}`)
+                .catch(function (error) {
+                    sendAPIError(response, error);
+                });
+            if (res) {
+                const apiData = (res as AxiosResponse<APIResponse<ProjectResponse>>).data.data;
+                // Get each project sample (sample hash and filename) from redis and find the
+                // corresponding sample (by hash) in the response from beebop_py api (containing cluster info etc) -
+                // combine data from both in response to client
+                const projectSamples = await store.getProjectSamples(projectId);
+                const responseSamples = [];
+                for (const sample of projectSamples) {
+                    const apiSample = apiData.samples.find(s => s.hash === sample.hash);
+                    if (!apiSample) {
+                        throw Error(`Sample with hash ${sample.hash} was not in API response`);
```

Comment:
when do we expect this to happen? logic error on our part, or is this something that could actually occur?